### PR TITLE
`azurerm_app_service_slot`  - support `application_logs.file_system`

### LINKED ISSUE #7007: Support for configuring the file system level for application logs
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In ARM I have this:

```json
{
            "name": "[concat(variables('app'), '/logs')]",
            "type": "Microsoft.Web/sites/config",
            "apiVersion": "2018-11-01",
            "location": "[variables('location')]",
            "dependsOn": [
                "[resourceId('Microsoft.Web/sites', variables('app'))]"
            ],
            "properties": {
                "applicationLogs": {
                    "fileSystem": {
                        "level": "Information"
                    }
                },
                "httpLogs": {
                    "fileSystem": {
                        "retentionInMb": 35,
                        "retentionInDays": 10,
                        "enabled": true
                    }
                }
            }
        }
```

In Terraform, I see that we have applicationLogs, blob, but I do not see fileSystem:level.

Here's the full ARM snippet for that section.
```json
{
  "id": "/subscriptions/subid/resourceGroups/azsdkdemo7rg/providers/Microsoft.Web/sites/azsdkdemo7app/config/logs",
  "name": "logs",
  "type": "Microsoft.Web/sites/config",
  "location": "West US",
  "tags": {
    "displayName": "app"
  },
  "properties": {
    "applicationLogs": {
      "fileSystem": {
        "level": "Information"
      },
      "azureTableStorage": {
        "level": "Off",
        "sasUrl": null
      },
      "azureBlobStorage": {
        "level": "Off",
        "sasUrl": null,
        "retentionInDays": null
      }
    },
    "httpLogs": {
      "fileSystem": {
        "retentionInMb": 35,
        "retentionInDays": 10,
        "enabled": true
      },
      "azureBlobStorage": {
        "sasUrl": null,
        "retentionInDays": 10,
        "enabled": false
      }
    },
    "failedRequestsTracing": {
      "enabled": false
    },
    "detailedErrorMessages": {
      "enabled": false
    }
  }
}
```

https://management.azure.com/subscriptions/subid/resourceGroups/azsdkdemo7rg/providers/Microsoft.Web/sites/azsdkdemo7app/config/logs?api-version=2018-02-01

It surfaces here in the Portal
image

https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/subid/resourceGroups/azsdkdemo7rg/providers/Microsoft.Web/sites/azsdkdemo7app/appservicelogs

Would be great to that section full covered by terraform.

azsdke2e

### PR DESCRIPTION
**Pull request by sschmeck**
Fixes #7007

### CODE REVIEW

File: azurerm/internal/services/web/tests/resource_arm_app_service_test.go [hunk 1]
```diff
@@ -3041,6 +3043,9 @@ resource "azurerm_app_service" "test" {
 
   logs {
     application_logs {
+      file_system {
+        level = "Warning"
+      }
```
Comment (katbyte) [inline azurerm/internal/services/web/tests/resource_arm_app_service_test.go:3048]: as there doesn't seem to be any other configuration for file system would this be better as
```suggestion
      file_system_level = "Warning"
```

**Comment by jongio** [comment]
Here's a scenario that I want to make sure is supported.

1. User settings file_system_level to Warning, and puts this in their TF file:
`file_system_level = Warning`
2. Time passes
3. User wants to turn it off.

User should be able to:
1. Remove `file_system_level = Warning` property altogether.
2. Set `file_syste_level = Off`


I'm assuming that the second option will work, but could you also test to see if the first option turns it off as well?

- Support: "Turn off file_system_level by omitting the property when file_system_level was previously set"

Thanks,
Jon

**Comment by katbyte** [comment]
@jongio that would require a default of off i think - @sschmeck does this default to off and would that make sense?

**Comment by sschmeck** [comment]
@katbyte Regarding the default value, I'm not sure it's necessary. 

I would expect the logging is turned off, if nothing is set explicitly. But I'm not able to test it. I'm not familiar with that kind configuration of configuration. I tried to check the extracted ARM templates for a App Service in the Azure Portal but couln't find the setting in the extracted template. :/ 

In the Azure Portal where you setup the App Service logs view, there is a tooltip for the _Application Logging (Filesystem)_ with the following tooltip - "Enable application logging to collect diagnostic traces from your web app code. You'll need to turn this on to enable the streaming log feature. This setting turns itself off after 12 hours."

Therefore the configuration seems not persistent.

**Comment by jongio** [comment]
LMK if you want me to test anything.

**Comment by jongio** [comment]
Thank you @sschmeck  and @katbyte !

**Pull request merged**